### ISSUE TITLE
[Image] Large "Portrait" images throws "Cannot process rendition original due to size" when loading

### ISSUE THREAD

**Issue by Semperdecus** [adobe/aem-core-wcm-components#1488]
## Bug Report
**Steps to reproduce**
Reproducible on a clean environment with latest core components installed. 

1. Create a page with the image core component
2. Drag and drop an image with size 4000x5000

**Current Behavior**
Image is loaded with the original rendition which is larger than 3840 and (with default settings) will cause the "Cannot process rendition XXX due to size" error. 

**Expected behavior/code**
Image is loaded with (by default) the 1280 rendition

**Environment**
- Adobe Experience Manager (6.5.5.0)
- Core Components version 2.13.2 & 2.15.3
- Java(TM) SE Runtime Environment (build 1.8.0_221-amd64)

**Possible Solutions**
The `getBestRendition` method in `AdaptiveImageServlet` shouldn't fallback to the "original" rendition if width is larger than the `DEFAULT_MAX_SIZE` AND height is larger than width. In our case we would like to return a 1920 rendition. 

**Additional context / Screenshots**
![error-image](https://user-images.githubusercontent.com/26483574/113146109-92dce380-922f-11eb-9a28-6e6044ead784.png)

**Comment by msagolj** [comment]
Hi @Semperdecus, 
- If the the image is wider then the largest rendition available it falls back to the original rendition to scale it down
- Taking the largest rendition that is smaller then the requested width would cause a stretching of the image
- The process size limit was introduced since processing large images on the instance itself can cause OOM
- You can change the default limit in the OSGI Config, see [here ](https://github.com/adobe/aem-core-wcm-components/blob/master/config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.servlets.AdaptiveImageServletMappingConfigurationFactory-coreimg.config )and [here](https://github.com/adobe/aem-core-wcm-components/blob/master/config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.servlets.AdaptiveImageServletMappingConfigurationFactory-img.config), or change it in the console itself:

![image](https://user-images.githubusercontent.com/37147400/113171420-bb70d780-9247-11eb-9423-bd964235beb0.png)

Hope I understood your issue correctly and above helps :)

**Comment by Semperdecus** [comment]
Thanks for your reply @msagolj 

I was aware that the maximum processed image size can be adjusted but this can cause memory issues so we're trying to avoid doing this. 

I'm not sure you understand the problem so I added a [diagram](https://user-images.githubusercontent.com/26483574/113318449-c5aad880-9310-11eb-851b-bdb6276fef6a.png) and made a [pull request](https://github.com/adobe/aem-core-wcm-components/pull/1496/commits/a6a921f2f4f7c16956731d613f6e029bd9c41249) to help illustrate the issue.

In short: there needs to be a fallback if the image exceeds the maximum processed image size (3840 by default). Having the image not load without giving an author a reason is not the optimal UX we're trying to achieve.

**Comment by bpauli** [comment]
Hi @Semperdecus,

thank you for the hands up and the effort to create the PR for an enhancement. I think this is a valid request but the provided solution to use the second last rendition is very case specific. As this logic would apply to all portrait formats it might not be the best fit for all users of the AdaptiveImageServlet.

The image processing and configuration in AEM is very flexible and the image component of this project make it even more complex. I wonder if it might be worth to introduce a new service which is responsible to pick the asset rendition from DAM and which can be customised on project level. The current implementation would then be the default implementation of that service which is shipped with the core component project but custom implementations with a higher service ranking would have precedence so it can be easily adapted to project requirements and specific configurations. WDYT?

**Comment by Semperdecus** [comment]
Hi @bpauli sorry for the delayed response.

To paraphrase, what you're thinking of doing is:
1. Splitting the the rendition functionality from AdaptiveImageServlet to some "RenditionPickingService"
2. Allow developers to use a higher service ranking to easily overwrite the default implementation

Looks like a great idea. It's definitely more flexible and scalable than the quick fix I suggested. I'm looking forward to seeing this realised

**Comment by joerghoh** [comment]
Another option would be to create a new rendition (via processing profiles) which has the same dimensions (or a bit larger) than the largest renditions you require in your frontend. In that case you never have to use the original renditions as a basis to downscale on the fly.

**Comment by vladbailescu** [comment]
Closing this issue as it's not something we want to fix/improve in the AdaptiveImageServlet since we now have the new web-optimized image delivery system: https://experienceleague.adobe.com/docs/experience-manager-core-components/using/developing/web-optimized-image-delivery.html?lang=en
